### PR TITLE
docs: Add some more examples, and fix some docs

### PR DESCRIPTION
**Pull request by KingDGrizzle**
**Please describe the changes this PR makes and why it should be merged:**
This could be counted as a WIP PR, because I would like to add some more examples on some more obscure functions, so that people know how they work.

Bring the docs main page up to date with the GitHub readme
Other small mistypes I noticed

I am waiting on people's opinions, and what they would like to see more of in the example section.
This way, we can show examples to people of what they would most commonly want

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [x] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by devsnek** [comment]
ffmpeg binaries don't need to be global

File: docs/examples/embed.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+client.on('ready', () => {
+  console.log('I am ready!');
+});
+
+client.on('message', message => {
+  // If the message is "how to embed"
+  if (message.content === 'how to embed') {
+    // We can create embeds using the MessageEmbed constructor
+    // Read more about all that you can do with the constructor in the docs
+    let embed = new MessageEmbed()
```
Comment (iCrawl) [inline docs/examples/embed.js]: `const`

File: docs/general/faq.md [hunk 2]
```diff
@@ -12,12 +12,12 @@ Update to Node.js 8.0.0 or newer.
   node-opus is greatly preferred, due to it having significantly better performance.
 
 ## How do I install FFMPEG?
-- **npm:** `npm install --save ffmpeg-binaries`
+- **npm:** `npm i -g ffmpeg-binaries`
```
Comment (iCrawl) [inline docs/general/faq.md]: No global, and keep it `install`

File: docs/general/faq.md [hunk 3]
```diff
@@ -12,12 +12,12 @@ Update to Node.js 8.0.0 or newer.
...
 - **Ubuntu 16.04:** `sudo apt install ffmpeg`
 - **Ubuntu 14.04:** `sudo apt-get install libav-tools`
-- **Windows:** See the [FFMPEG section of AoDude's guide](https://github.com/bdistin/OhGodMusicBot/blob/master/README.md#download-ffmpeg).
+- **Windows:** Use the npm version or see the [FFMPEG section of AoDude's guide](https://github.com/bdistin/OhGodMusicBot/blob/master/README.md#download-ffmpeg).
 
 ## How do I set up node-opus?
-- **Ubuntu:** Simply run `npm install node-opus`, and it's done. Congrats!
-- **Windows:** Run `npm install --global --production windows-build-tools` in an admin command prompt or PowerShell.
-  Then, running `npm install node-opus` in your bot's directory should successfully build it. Woo!
+- **Ubuntu:** Simply run `npm i node-opus`, and it's done. Congrats!
```
Comment (iCrawl) [inline docs/general/faq.md]: Same as above.

File: docs/general/faq.md [hunk 4]
```diff
@@ -12,12 +12,12 @@ Update to Node.js 8.0.0 or newer.
...
+- **Windows:** Run `npm i -g --production windows-build-tools` in an admin command prompt or PowerShell.
+  Then, running `npm i node-opus` in your bot's directory should successfully build it. Woo!
```
Comment (iCrawl) [inline docs/general/faq.md]: Same as above.

File: docs/general/faq.md [hunk 5]
```diff
@@ -12,12 +12,12 @@ Update to Node.js 8.0.0 or newer.
...
 - **Ubuntu 14.04:** `sudo apt-get install libav-tools`
-- **Windows:** See the [FFMPEG section of AoDude's guide](https://github.com/bdistin/OhGodMusicBot/blob/master/README.md#download-ffmpeg).
+- **Windows:** Use the npm version or see the [FFMPEG section of AoDude's guide](https://github.com/bdistin/OhGodMusicBot/blob/master/README.md#download-ffmpeg).
 
 ## How do I set up node-opus?
-- **Ubuntu:** Simply run `npm install node-opus`, and it's done. Congrats!
-- **Windows:** Run `npm install --global --production windows-build-tools` in an admin command prompt or PowerShell.
-  Then, running `npm install node-opus` in your bot's directory should successfully build it. Woo!
+- **Ubuntu:** Simply run `npm i node-opus`, and it's done. Congrats!
+- **Windows:** Run `npm i -g --production windows-build-tools` in an admin command prompt or PowerShell.
```
Comment (iCrawl) [inline docs/general/faq.md]: Same as above, but can stay global.

File: docs/general/welcome.md [hunk 6]
```diff
@@ -32,22 +30,23 @@ discord.js is a powerful [node.js](https://nodejs.org) module that allows you to
 **Node.js 8.0.0 or newer is required.**  
 Ignore any warnings about unmet peer dependencies, as they're all optional.
 
-Without voice support: `npm install discord.js --save`  
-With voice support ([node-opus](https://www.npmjs.com/package/node-opus)): `npm install discord.js node-opus --save`  
-With voice support ([opusscript](https://www.npmjs.com/package/opusscript)): `npm install discord.js opusscript --save`
+Without voice support: `npm i discord.js`
```
Comment (iCrawl) [inline docs/general/welcome.md]: No shortening to `i`

File: docs/general/welcome.md [hunk 7]
```diff
@@ -32,22 +30,23 @@ discord.js is a powerful [node.js](https://nodejs.org) module that allows you to
 **Node.js 8.0.0 or newer is required.**  
 Ignore any warnings about unmet peer dependencies, as they're all optional.
 
-Without voice support: `npm install discord.js --save`  
-With voice support ([node-opus](https://www.npmjs.com/package/node-opus)): `npm install discord.js node-opus --save`  
-With voice support ([opusscript](https://www.npmjs.com/package/opusscript)): `npm install discord.js opusscript --save`
+Without voice support: `npm i discord.js`  
+- With voice support ([node-opus](https://www.npmjs.com/package/node-opus)): `npm i discord.js node-opus`
```
Comment (iCrawl) [inline docs/general/welcome.md]: See above.

File: docs/general/welcome.md [hunk 8]
```diff
@@ -32,22 +30,23 @@ discord.js is a powerful [node.js](https://nodejs.org) module that allows you to
 **Node.js 8.0.0 or newer is required.**  
 Ignore any warnings about unmet peer dependencies, as they're all optional.
 
-Without voice support: `npm install discord.js --save`  
-With voice support ([node-opus](https://www.npmjs.com/package/node-opus)): `npm install discord.js node-opus --save`  
-With voice support ([opusscript](https://www.npmjs.com/package/opusscript)): `npm install discord.js opusscript --save`
+Without voice support: `npm i discord.js`  
+- With voice support ([node-opus](https://www.npmjs.com/package/node-opus)): `npm i discord.js node-opus`  
+- With voice support ([opusscript](https://www.npmjs.com/package/opusscript)): `npm i discord.js opusscript`
```
Comment (iCrawl) [inline docs/general/welcome.md]: See above.

File: docs/general/welcome.md [hunk 9]
```diff
@@ -32,22 +30,23 @@ discord.js is a powerful [node.js](https://nodejs.org) module that allows you to
...
-Using opusscript is only recommended for development environments where node-opus is tough to get working.
-For production bots, using node-opus should be considered a necessity, especially if they're going to be running on multiple servers.
+The preferred audio engine is `node-opus`, as it performs significantly better than `opusscript`. When both are available, discord.js will automatically choose `node-opus`.
+Using `opusscript` is only recommended for development environments where node-opus is tough to get working.
+For production bots, using `node-opus` should be considered a necessity, especially if they're going to be running on multiple servers.
 
 ### Optional packages
-- [bufferutil](https://www.npmjs.com/package/bufferutil) to greatly speed up the WebSocket when *not* using uws (`npm install bufferutil --save`)
-- [erlpack](https://github.com/hammerandchisel/erlpack) for significantly faster WebSocket data (de)serialisation (`npm install discordapp/erlpack --save`)
+- [zlib-sync](https://www.npmjs.com/package/zlib-sync) for significantly faster WebSocket data inflation (`npm i zlib-sync`)
```
Comment (iCrawl) [inline docs/general/welcome.md]: See above.

File: docs/general/welcome.md [hunk 10]
```diff
@@ -32,22 +30,23 @@ discord.js is a powerful [node.js](https://nodejs.org) module that allows you to
...
-For production bots, using node-opus should be considered a necessity, especially if they're going to be running on multiple servers.
+The preferred audio engine is `node-opus`, as it performs significantly better than `opusscript`. When both are available, discord.js will automatically choose `node-opus`.
+Using `opusscript` is only recommended for development environments where node-opus is tough to get working.
+For production bots, using `node-opus` should be considered a necessity, especially if they're going to be running on multiple servers.
 
 ### Optional packages
-- [bufferutil](https://www.npmjs.com/package/bufferutil) to greatly speed up the WebSocket when *not* using uws (`npm install bufferutil --save`)
-- [erlpack](https://github.com/hammerandchisel/erlpack) for significantly faster WebSocket data (de)serialisation (`npm install discordapp/erlpack --save`)
+- [zlib-sync](https://www.npmjs.com/package/zlib-sync) for significantly faster WebSocket data inflation (`npm i zlib-sync`) 
+- [erlpack](https://github.com/discordapp/erlpack) for significantly faster WebSocket data (de)serialisation (`npm i discordapp/erlpack`)
```
Comment (iCrawl) [inline docs/general/welcome.md]: See above.

File: docs/topics/voice.md [hunk 11]
```diff
@@ -4,10 +4,10 @@ Voice in discord.js can be used for many things, such as music bots, recording o
 In discord.js, you can use voice by connecting to a `VoiceChannel` to obtain a `VoiceConnection`, where you can start streaming and receiving audio.
 
 To get started, make sure you have:
-* ffmpeg - `npm install ffmpeg-binaries`
+* FFmpeg - `npm i -g ffmpeg-binaries`
```
Comment (iCrawl) [inline docs/topics/voice.md]: No global, keep it `install`

File: docs/topics/voice.md [hunk 12]
```diff
@@ -4,10 +4,10 @@ Voice in discord.js can be used for many things, such as music bots, recording o
...
 * an opus encoder, choose one from below:
-  * `npm install opusscript`
-  * `npm install node-opus`
+  * `npm i opusscript`
```
Comment (iCrawl) [inline docs/topics/voice.md]: See above.

File: docs/topics/voice.md [hunk 13]
```diff
@@ -4,10 +4,10 @@ Voice in discord.js can be used for many things, such as music bots, recording o
...
+  * `npm i node-opus`
```
Comment (iCrawl) [inline docs/topics/voice.md]: See above.

**Comment by iCrawl** [comment]
Also don't bring in random doc changes to a PR that handles examples and the docs pages.

File: docs/general/welcome.md [hunk 14]
```diff
@@ -10,15 +10,13 @@
...
     <a href="https://travis-ci.org/hydrabolt/discord.js"><img src="https://travis-ci.org/hydrabolt/discord.js.svg" alt="Build status" /></a>
     <a href="https://david-dm.org/hydrabolt/discord.js"><img src="https://img.shields.io/david/hydrabolt/discord.js.svg?maxAge=3600" alt="Dependencies" /></a>
+    <a href="https://www.patreon.com/discordjs"><img src="https://img.shields.io/badge/donate-patreon-F96854.svg" alt="Patreon" /></a>
   </p>
   <p>
     <a href="https://nodei.co/npm/discord.js/"><img src="https://nodei.co/npm/discord.js.png?downloads=true&stars=true" alt="NPM info" /></a>
   </p>
 </div>
 
-# Welcome!
```
Comment (iCrawl) [inline docs/general/welcome.md]: ?

Comment (KingDGrizzle) [reply docs/general/welcome.md]: Most of the changes to welcome.md were made cause I updated it from the GitHub readme. I'll do your requests today, once I get home. :smiley:

**Comment by KingDGrizzle** [comment]
Closing this, to solve all issues, as well as split up actual doc fixes from adding more examples.

**Pull request closed without merging**